File: 021_Java/fizzbuzz.java
// This is a comment.

public class fizzbuzz {

    private int max_;

    //constructor
    public fizzbuzz(int max) {
        this.max_ = max;
    }

    public void run_() {
        for (int i = 1; i <= this.max_; ++i) {
            printNum_(i);
        }
    }

    public void printNum_(int i) {
        if (i % 15 == 0) {
            System.out.println("FizzBuzz");
        } else if (i % 3 == 0) {
            System.out.println("Fizz");
        } else if (i % 5 == 0) {
            System.out.println("Buzz");
        } else {
            System.out.println(i);
        }
    }

    public static void main(String[] args) {
        fizzbuzz fb = new fizzbuzz(100);
        fb.run_();
    }

}

File: 014_Perl/fizzbuzz.pl
# This is a comment.

#To declare a variable, you can also use the `our` keyword.
#`my` declares a local variable whilst `our` declares a global variable.
#See |https://stackoverflow.com/questions/845060/what-is-the-difference-between-my-and-our-in-perl| for the detail.

sub fizzbuzz {

    my $i = $_[0];
    my $is_fizz = ($i % 3 == 0);
    my $is_buzz = ($i % 5 == 0);

    return 'FizzBuzz' if ($is_fizz && $is_buzz);
    return 'Fizz' if ($is_fizz);
    return 'Buzz' if ($is_buzz);
    $i; #This last `return` can be omitted.

}

for my $i (1..100) {
    print fizzbuzz($i)."\n";
}

